docker/Dockerfile

```diff
@@ -23,16 +24,18 @@ COPY . /code/
 # zipkin-builder - An image that caches build repositories (.m2, .npm) to speed up subsequent builds.
 #####
 
-FROM openzipkin/zipkin-builder as built
+FROM openzipkin/zipkin-builder as install
 
-COPY --from=scratch /code /code
+WORKDIR /install
 
-WORKDIR /code
+# Conditions aren't supported in Dockerfile instructions, so we copy source even if it isn't used.
+COPY --from=scratch /code /code
 
-# Same command as we suggest in README.md: disables JDK enforcement only needed in tests
-RUN ./mvnw -q --batch-mode -DskipTests -Denforcer.fail=false --also-make -pl main package
+# This will either be "master" or a real version ex. "2.4.5"
+ARG RELEASE_VERSION=master
+COPY docker/install.sh /tmp/
```

Comment:
except the build arg here, this is the same look and feel as building our demo images, such as cassandra